modules/task/task.go

```diff
@@ -75,6 +79,10 @@ func newAsyncModule(
 	_globalBackendMu.Lock()
 	_globalBackend = backend
 	_globalBackendMu.Unlock()
+	// Register context implementation with the encoder so it can be used in async functions
+	if err := _globalBackend.Encoder().Register(context.Background()); err != nil {
```

Comment:
Context.Background is of type emptyCtx. I've counted at least 3 more context types in the source code: https://golang.org/src/context/context.go?s=11899:11974#L311

I think it is really bad idea to depend on implementation on standard packages..